esmvalcore/preprocessor/_regrid.py

```diff
@@ -433,14 +434,17 @@ def _vertical_interpolate(cube, src_levels, levels, interpolation,
         src_levels.points, cube.shape, cube.coord_dims(src_levels))
 
     # force mask onto data as nan's
-    if np.ma.is_masked(cube.data):
-        cube.data[cube.data.mask] = np.nan
+    if da.ma.getmaskarray(cube.core_data()).any():
+        if cube.has_lazy_data():
+            cube.data = da.ma.filled(cube.core_data(), np.nan)
+        else:
+            cube.data[cube.data.mask] = np.nan
```

Comment:
```suggestion
cube.data = da.ma.filled(cube.core_data(), np.nan)
```
This is what I mean, it looks like this works and I think it's better because we leave all the if/else checks to `da.ma.filled`

Reply:
ahaa! now I get it - didn't know that thing was powerful enough to account for the realized data and fill it with nans if needed, cheers dude!